docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/utils/PortUtils.java

```diff
@@ -1,70 +1,116 @@
 package com.nirima.jenkins.plugins.docker.utils;
 
+import com.trilead.ssh2.Connection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.IOException;
 import java.net.Socket;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static shaded.com.google.common.base.Preconditions.checkState;
 
 public class PortUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PortUtils.class);
 
-    private static final int RETRIES = 10;
-    private static final int WAIT_TIME_MS = 2000;
+    private final String host;
+    private final int port;
 
-    public static boolean isPortAvailable(String host, int port) {
-        Socket socket = null;
-        boolean available = false;
-        try {
-            socket = new Socket(host, port);
-            available = true;
+    private int retries = 10;
+    private int sshTimeoutMillis = (int) SECONDS.toMillis(2);
```

Comment:
`ConnectionInfo connect(ServerHostKeyVerifier verifier, int connectTimeout, int readTimeout, int kexTimeout)` - that's the main cause of int cast